File: 02_sort/02-03_insertion-sort.js
function insertion_sort(array) {
  for (let i = 1; i < array.length; i++ ) {
    let j;
    const val = array[i]
    for (j = i; j > 0 && array[j - 1] > val; j-- ) { // 삽입할 위치 찾기
      array[j] = array[j - 1]
    }
    array[j] = val
  }

  return array
}

const unsorted_array = [30, 20, 40, 35, 5, 10, 45, 50, 25, 15]
const sorted_array = insertion_sort(unsorted_array)

console.log(sorted_array)

File: 02_sort/02-03_insertion-sort.py
def insertion_sort(array):
    for i in range(1, len(array) ):
        val = array[i]
        for j in range(i, 0, -1):
            if array[j - 1] > val:
                array[j - 1], array[j] = array[j], array[j - 1]

    return array


unsorted_array = [30, 20, 40, 35, 5, 10, 45, 50, 25, 15]
sorted_array = insertion_sort(unsorted_array)

print(sorted_array)
